openwebui_installer/downloader.py

```
@@ -0,0 +1,66 @@
+                return dest
+
+        try:
+            response = self.session.get(url, stream=True, timeout=30)
+            response.raise_for_status()
+            os.makedirs(os.path.dirname(dest), exist_ok=True)
+            with open(dest, "wb") as f:
+                for chunk in response.iter_content(chunk_size=8192):
+                    if chunk:
+                        f.write(chunk)
+        except Exception as exc:  # pragma: no cover - real network errors
+            raise DownloadError(f"Failed to download {url}: {exc}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)